test/components/date-picker.spec.tsx

```diff
@@ -24,72 +25,85 @@ describe('The DatePicker Component', function() {
-            async function() {
-                const {select, waitForDom} = clientRenderer.render(<DatePickerDemo />);
+            const datePickerInput = select(datePickerInputId);
+            trigger.change(datePickerInput!, '2017/02/01');
+            simulate.keyDown(datePickerInput, { keyCode: keycode('enter') });
 
-                const datePickerInput = select(datePickerInputId);
-                triggerChange(datePickerInput!, '2017/02/01');
-                simulate.keyDown(datePickerInput, { keyCode: keycode('enter') });
+            await waitForDom(() => expect(select(currentDate)).to.have.text('Wed Feb 01 2017'));
+        });
 
-                await waitForDom(() => expect(select(currentDate)).to.have.text('Wed Feb 01 2017'));
-            }
-        );
+        it('clicks on the input, picks a date from the dropdown, ' +
+            'and then expects the dropdown to close and the date to appear in the input', async () => {
+            const {select, waitForDom} = clientRenderer.render(<DatePickerDemo value={JANUARY_FIRST} />);
+            const datePickerInput = select(datePickerInputId);
 
-        it(
-            `clicks on the input, picks a date from the dropdown,
-            and then expects the dropdown to close and the date to appear in the input`,
-            async function() {
-                const {select, waitForDom} = clientRenderer.render(<DatePickerDemo value={JANUARY_FIRST} />);
-                const datePickerInput = select(datePickerInputId);
+            await waitForDom(() => expect(select(datePickerDropdownId)).to.be.absent());
 
-                await waitForDom(() => expect(select(datePickerDropdownId)).to.be.absent());
+            simulate.focus(datePickerInput);
 
-                simulate.focus(datePickerInput);
+            await waitForDom(() => expect(select(datePickerDropdownId)).to.be.present());
 
-                await waitForDom(() => expect(select(datePickerDropdownId)).to.be.present());
+            simulate.mouseDown(select('DAY_4'));
 
-                simulate.mouseDown(select('DAY_4'));
+            await waitForDom(() => {
+                expect(select(datePickerDropdownId)).to.be.absent();
+                expect(select(currentDate)).to.have.text('Wed Jan 04 2017');
+            });
+        });
+    });
 
-                await waitForDom(() => {
-                    expect(select(datePickerDropdownId)).to.be.absent();
-                    expect(select(currentDate)).to.have.text('Wed Jan 04 2017');
-                });
-            }
-        );
+    it('should only call onChange once', async () => {
+        const onChange = sinon.spy();
+        const {select, waitForDom} = clientRenderer.render(<DatePicker onChange={onChange} />);
+        const datePickerInput = select(datePickerInputId);
+
+        trigger.change(datePickerInput!, '2017/02/01');
+        simulate.blur(datePickerInput);
+
+        await waitForDom(() => expect(onChange.calledOnce));
```

Comment:
I totally thought I asserted it that way... changed it to the method you mentioned.